test_integration/lib/utils/encoders.dart

```diff
@@ -1,5 +1,17 @@
 import 'package:ably_flutter/ably_flutter.dart';
 
+Map<String, dynamic> encodeMessageExtras(MessageExtras extras) => extras == null
```

Comment:
Should this be a method on `MessageExtras` instead of a function?, so usage will be `message.extras.toMap()`. Somewhat related to renaming `message.extras.extras` into something more readable. Then in the tests, no extra functions are needed

Reply:
Not sure we should do that if there is no spec item for that! 🤔 

@QuintinWillison what's your opinion?

Reply:
It's not a feature spec point, its an implementation detail related to clean code/ encapsulation.

Reply:
Perhaps a compromise would be to make this an [extension method](https://dart.dev/guides/language/extension-methods), defined by the test suite. We would not want to be exposing this as public API on the SDK, especially if it's not defined in the feature spec..

Reply:
I'd say we leave it at its existing implementation for now. Please create an issue if this change could have a positive impact on performance or readability.